src/main/java/tuteez/model/tag/Lesson.java

```diff
@@ -0,0 +1,92 @@
+package tuteez.model.tag;
+
+
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
+import static java.util.Objects.requireNonNull;
+import static tuteez.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Lesson in the address book.
+ * Guarantees: immutable; day and time are valid as declared in {@link #isValidLesson(String, String)}
+ */
+public class Lesson {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Days should be one of the following: Monday, Tuesday, Wednesday, Thursday, Friday, Saturday, Sunday (case insensitive). "
+            + "Lesson time should be in the format HHMM-HHMM (24-hour format). "
+            + "Start time must be before end time.";
+    private static final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HHmm");
+    private static final String VALID_TIME_RANGE_REGEX = "([01]?[0-9]|2[0-3])[0-5][0-9]-([01]?[0-9]|2[0-3])[0-5][0-9]";
+    public final String day;
+    public final String timeRange;
+
+    /**
+     * Constructs a {@code Lesson}.
+     *
+     * @param day A valid day.
+     * @param timeRange A valid lesson time range (HHMM-HHMM).
+     */
+    public Lesson(String day, String timeRange) {
```

Comment:
Would it be better to take in an entire `String` with both the day and time? 
i.e. The responsibility of splitting the `String` into `day` and `timeRange` falls on the constructor